controllers/nstemplateset/namespaces.go

```diff
@@ -76,13 +76,72 @@ func (r *namespacesManager) ensure(logger logr.Logger, nsTmplSet *toolchainv1alp
 func (r *namespacesManager) ensureNamespace(logger logr.Logger, nsTmplSet *toolchainv1alpha1.NSTemplateSet, tierTemplate *tierTemplate, userNamespace *corev1.Namespace) error {
 	logger.Info("ensuring namespace", "namespace", tierTemplate.typeName, "tier", nsTmplSet.Spec.TierName)
 
-	// create namespace before created inner resources because creating the namespace may take some time
+	namespaceNeedsUpdate := false
```

Comment:
maybe 
```suggestion
	createOrUpdateNamespace := false
```
since it applies also to creation

Reply:
Done in 035cdca7d68d85445177585f27f3d28cdadb2dcd